### ISSUE TITLE
ECS deploys broken in release-1.19.x-latest-validated

### ISSUE THREAD

**Issue by awsiv** [spinnaker/spinnaker#5551]
### Issue Summary:
After upgrading to `release-1.19.x-latest-unvalidated` I was unable to deploy an ECS service

### Cloud Provider(s):
ecs

### Environment:
local debian

### Feature Area:
clouddriver, ecs

### Description:
After upgrading to `release-1.19.x-latest-unvalidated`, ECS deploy (test-v009) was taking forever.

Clouddriver logs show the following log:
```
Mar 06 22:52:06 ip-172-28-10-142 clouddriver[25314]: 2020-03-06 22:52:06.410  INFO 25314 --- [0.1-7002-exec-7] c.n.s.c.e.p.v.EcsServerClusterProvider   : No ECS Server Groups were found with the name test-v009
Mar 06 22:52:06 ip-172-28-10-142 gate[25305]: 2020-03-06 22:52:06.420  INFO 25305 --- [-serverGroups-9] c.n.s.g.s.internal.ClouddriverService    : <--- HTTP 404 http://localhost:7002/applications/test/serverGroups/stage-ecs/us-east-2/test-v009?includeDetails=false (191ms)

```

looks as if the newly created servergroup never gets into redis

```
redis> keys *test-v009*
(empty list or set)
(1.26s)
```

### Steps to Reproduce:
Upgrade to `release-1.19.x-latest-unvalidated`  and start an ECS deploy

### Additional Details:
Some weirdness in Deck as well: items in the cluster view disappear periodically

<img width="790" alt="Screenshot 2020-03-06 at 23 41 30" src="https://user-images.githubusercontent.com/1265825/76129701-5c22a280-6008-11ea-82aa-78c4e14f3e0d.png">
<img width="814" alt="Screenshot 2020-03-06 at 23 39 46" src="https://user-images.githubusercontent.com/1265825/76129705-5d53cf80-6008-11ea-89db-3c4e01eccc96.png">

**Comment by awsiv** [comment]
Possibly related changes:
- https://github.com/spinnaker/clouddriver/pull/4274
- https://github.com/spinnaker/spinnaker/issues/4495

**Comment by awsiv** [comment]
Duplicate: https://github.com/spinnaker/spinnaker/issues/5528

**Comment by allisaurus** [comment]
@awsiv I was able to deploy after upgrading to `release-1.19.x-latest-validated`. Could you provide your (redacted) pipeline JSON so I can try to reproduce?

I do see you list trying `release-1.19.x-latest-**unvalidated**`, so if you haven't given `validated` a shot can you try that and let me know if you see the same results?

**Comment by allisaurus** [comment]
potentially related to https://github.com/spinnaker/spinnaker/issues/5528 - log msg is similar
(I see you mentioned this issue previously but I don't see a link in the current description)

**Comment by awsiv** [comment]
@allisaurus thanks, will try `validated` tag

Also #5528 looked related at first glance but the deck behavior was also concerning so removed my comment (I should stop doing that :) )

**Comment by allisaurus** [comment]
ah, gotcha! yes the deck behavior is strange too, thanks for including that info as well.

**Comment by allisaurus** [comment]
@awsiv can you also describe where in the deployment execution it was hanging? Was it during the `wait for up instances` deployment task, as in #5528 ?

**Comment by awsiv** [comment]
@allisaurus yep.. it appeared to hang at the same place .. with similar error log as you pointed out earlier

**Comment by allisaurus** [comment]
@awsiv could you tell me how large (in terms of total ECS services & tasks, not just those run by spinnaker) your account is in the the region where you're trying to deploy? We've [seen issues before](https://github.com/spinnaker/spinnaker/issues/4495) where rate limiting can stall deployments and I'm trying to figure out if this could be a factor.

It would also be a big help to see your (redacted) pipeline JSON so I can try and repro the error.

**Comment by awsiv** [comment]
@allisaurus We have just started with ECS and fargate and as such not many services there.

~30 services spread across 4 accounts with 2 clusters each.
Most of our services are on EC2 of which ~90% are deployed from spinnaker

My observations:
- I tested by increasing the rate limits - this didn't help (in fact, having higher limit on aws provider than ecs provider made things worse for deck)
- The issues is more with the caching IMO, the cluster view doesn't show existing cluster/instances at all. Security groups are not shown as well and as mentioned above ecs resources do not end up in redis.
- however, looking at the logs, I can see that caching is active
```
Mar 10 22:27:36 ip-172-28-10-142 clouddriver[2944]: 2020-03-10 22:27:36.351  INFO 2944 --- [ecutionAction-2] c.n.s.c.e.p.agent.TaskCachingAgent       : Caching 26 tasks in account1/us-east-2/TaskCachingAgent
Mar 10 22:27:36 ip-172-28-10-142 clouddriver[2944]: 2020-03-10 22:27:36.351  INFO 2944 --- [ecutionAction-2] c.n.s.c.e.p.agent.TaskCachingAgent       : Caching 1 ECS clusters in account1/us-east-2/TaskCachingAgent
Mar 10 22:27:36 ip-172-28-10-142 clouddriver[2944]: 2020-03-10 22:27:36.357  INFO 2944 --- [ecutionAction-2] c.n.s.c.e.p.agent.TaskCachingAgent       : Evicting 1 tasks in account1/us-east-2/TaskCachingAgent
Mar 10 22:27:38 ip-172-28-10-142 clouddriver[2944]: 2020-03-10 22:27:38.352  INFO 2944 --- [cutionAction-52] c.n.s.c.e.p.agent.ServiceCachingAgent    : Caching 14 services in account1/us-east-2/ServiceCachingAgent
Mar 10 22:27:38 ip-172-28-10-142 clouddriver[2944]: 2020-03-10 22:27:38.353  INFO 2944 --- [cutionAction-52] c.n.s.c.e.p.agent.ServiceCachingAgent    : Caching 1 ECS clusters in account1/us-east-2/ServiceCachingAgent

Mar 10 22:27:53 ip-172-28-10-142 clouddriver[2944]: 2020-03-10 22:27:53.212  INFO 2944 --- [cutionAction-12] reCachingAgent$EcsAwsAwareCacheExecution : Setting AWS Provider: com.netflix.spinnaker.clouddriver.aws.provider.AwsProvider

```
- but again, it is not active in another account the serviceCaching agent is active but appears to be skipping caching *services*. It appears to be caching clusters, tasks and others though 😕 . The only difference I can see is that service discovery is enabled for one of the services in this account.

- I also tested by setting redis caching interval back to default 30sec. This didn't help as well

Our pipelines are very simple: jenkins trigger -> find image -> deploy -> jenkins job to update autoscaling policy. 

Hopefully this helped, please let me know if you need further info. You can also find me on spinnaker slack 🙂

**Comment by allisaurus** [comment]
(adding info from slack discussion) FYI, sounds like the deck issues are also seen with 1.18.5

**Comment by awsiv** [comment]
hmm.. this looks very much related to what I'm experiencing : https://github.com/spinnaker/spinnaker/issues/4500

**Comment by allisaurus** [comment]
@spinnakerbot add-label bug

**Comment by awsiv** [comment]
tested the latest `release-1.19.x-latest-validated` - this issue is now resolved! 💯 

Thanks @allisaurus !

**Comment by awsiv** [comment]
oh.. now I'm seeing deployments stuck in:

```
Mar 15 17:40:53 ip-... orca[21090]: 2020-03-15 17:40:53.872 DEBUG 21090 --- [    handlers-18] c.n.s.o.c.t.i.WaitForUpInstancesTask     : [email] Unable to determine initial target capacity (serverGroup: ecs:us-east-1:testapp-v024, executionId: 01E3FM6GZ2R3P8R9SJXMM5GWJ1)

```

ECS cluster deployed with a single target group. If I enable Instance Health only, this works:
<img width="580" alt="image" src="https://user-images.githubusercontent.com/1265825/76707165-46d70380-66ed-11ea-96de-54286e3208f5.png">

Looks like spinnaker is unable to get health status from target group. Any suggestions?

**Comment by allisaurus** [comment]
Glad to hear you're seeing some improvement @awsiv ! Though I'd like to get to the bottom of things hanging when using load balancer health.

> BTW, it this available in 1.19.0 ?

https://github.com/spinnaker/clouddriver/pull/4417 will be available in 1.19.1, which I understand will be out very soon. 

re: closing this - in https://github.com/spinnaker/spinnaker/issues/5528 it sounds like deployments are hanging with either load balancer or cloud provider health checks, so I'm ok with keeping this open until we understand what's happening in your case.

**Comment by awsiv** [comment]
@allisaurus I upgraded to `1.19.1` today and deploy seems to be working fine 🎉 Deployments are no londer stuck in ` Wait For Up Instances` 

so it looks like a one-off issue. Thanks for you help in solving this issue! 💯  👍
